torch/optim/lr_scheduler.py

```diff
@@ -435,6 +457,45 @@ def _get_closed_form_lr(self):
                 for base_lr in self.base_lrs]
 
 
+class PolyLR(_LRScheduler):
+    """Decays the learning rate of each parameter group polynomially in given period of time.
+    When pow=1, lr decays linearly. When pow>1, lr drops fast then slowly converges to the end. When pow<1, otherwise.
+    pow<0 is forbidden because it would be an increase curve rather than a decrease curve.
+
+    Args:
+        optimizer (Optimizer): Wrapped optimizer.
+        pow (float): power of polynomial.
+        max_epoch (int): time period for polynomial annealing.
+        min_lrs (float or list): a float indicating the end of annealing lr,
+            or a list of float, one for each group in optimizer.param_groups 
+    """
+    def __init__(self, optimizer, pow, max_epoch, min_lrs=0, last_epoch=-1):
+        if pow < 0:
+            raise ValueError("Expected non-negative pow.")
+
+        self.pow = pow
+        self.max_epoch = max_epoch
+        if not isinstance(min_lrs, list) and not isinstance(min_lrs, tuple):
+            self.min_lrs = [min_lrs] * len(optimizer.param_groups)
+
+        super().__init__(optimizer, last_epoch=last_epoch)
+
+    def get_lr(self):
+        if self.last_epoch == 0:
+            return self.base_lrs
+
+        base = max(0, 1 - 1 / (self.max_epoch - self.last_epoch - 1))
+        coeff = base ** coeff
+        return [(lr - min_lr) * coeff + min_lr
+                for group, min_lr in zip(self.optimizer.param_groups, self.min_lrs)]
```

Comment:
We now make sure that schedulers are chainable were possible, see #26423. This essentially overrides the current value of the scheduler. Is there a way to modify the formula so that the following holds?

> lr(epoch) = lr(epoch-1) * factor

Reply:
Oops, `lr` above should be `group['lr']`. fixed.